### PR TITLE
Fix/password not present after tapping back

### LINKED ISSUE #20598: [Bug]: Tap and hold to reveal private key is not presented after tapping back arrow and then tapping show private key a second time
### Describe the bug

After PR 20109, there is some inconsistent behavior with the tap and hold to reveal button after tapping the back arrow on the private key pop-up. If I use back, and then tap "Show private key" a second time, the tap and hold to reveal does not appear and instead presents the private key value. For SRP reveal, there is no back arrow on extension. On iOS mobile, when tapping the back arrow from "Show private key" returns the user to wallet view instead of account details.

### Steps to reproduce

1. Setup a wallet and be at wallet view
2. Select accounts drop-down
3. Select kebab menu next to an HD account
4. Select "Account details"
5. Select "Show private key"
6. Provide correct password and select "confirm"
7. Click and hold to reveal private key
8. Select the back arrow and note that you are returned to "Account details"
9. Select "Show private key"
10. Note that there is no prompt for the password or to tap and hold, the private key is simply presented
11. This can be observed on both Chrome and Firefox and is not consistent with the mobile client behavior.

### Error messages or log output

_No response_

### Version

10.34.5

### Build type

None

### Browser

Chrome, Firefox

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

Video of Firefox: https://recordit.co/G1tIIcPoDx
Video of iOS mobile client: https://recordit.co/MDnNHXm80O

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**

This PR fix the issue #20598 , 
The fix will make sure password authentication will popup every time user click "Show private key" in account detail screen in Metamask Extension. this behaviour will be same as Metamask mobile behaviour.

## **Related issues**

Fixes: #20598

## **Manual testing steps**

1. Setup a wallet and be at wallet view
2. Select accounts drop-down
3. Select kebab menu next to an HD account
4. Select "Account details"
5. Select "Show private key"
6. Provide correct password and select "confirm"
7. Click and hold to reveal private key
8. Select the back arrow and note that you are returned to "Account details"
9. Select "Show private key"
10. Password authentication popup should shown again to ask user input their password.
11. Provide correct password and select "confirm"
12. Click and hold to reveal private key

### **Before**

https://recordit.co/G1tIIcPoDx

### **After**

https://github.com/MetaMask/metamask-extension/assets/7315988/a93323aa-4127-4d86-ac53-36b1b02c27e5

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by dawnseeker8** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.tsx [hunk 1]
```diff
@@ -14,9 +14,10 @@ import {
   Modal,
   ModalOverlay,
   Text,
+  ModalContent,
+  ModalHeader,
 } from '../../../component-library';
-import { ModalContent } from '../../../component-library/modal-content/deprecated';
```
Comment (dawnseeker8) [inline ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.tsx:18]: Remove two deprecated component and replace with new `ModalContent` and `ModalHeader`

**Review by HowardBraham — CHANGES_REQUESTED** [review]
The margins are not correct on the dialog that says "Keep your private key safe. Your Private Key provides full access to your wallet and funds."

Please look at the layout before this change and match it.

File: ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.tsx (hunk 1 above)
Comment (dawnseeker8) [reply ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.tsx:18]: Due to styling issue, revert the code back to use `deprecated` component.

**Comment by dawnseeker8** [comment]
hi, @HowardBraham i have revert the code in `hold-to-reveal-modal.tsx`, and now styling is correct again. please review again when you have time, thank you very much!

**Review by HowardBraham — APPROVED** [review]
I re-implemented the new Design System components and fixed the margins

**Review by owencraston — APPROVED** [review]

**Pull request merged**